go/vt/vttablet/onlineddl/executor.go

```diff
@@ -1102,6 +1108,11 @@ func (e *Executor) cutOverVReplMigration(ctx context.Context, s *VReplStream, sh
 	time.Sleep(100 * time.Millisecond)
 
 	if shouldForceCutOver {
+		// We should only proceed with forceful cut over if there is no pending atomic transaction for the table.
+		// This will help in keeping the atomicity guarantee of a prepared transaction.
+		if !e.IsPreparedPoolEmpty(onlineDDL.Table) {
```

Comment:
Followup to offline discussion, it's fine to spend some time here (a few hundred milliseconds?) waiting for the prepared pool to be empty.